### PR TITLE
Chore/cdd 2283 a

### PR DESCRIPTION
**Pull request by luketowell**
# Description

This PR includes the following:
- Removal of the date_posted attribute from all models apart from the whats new parent and the whats new challenge pages
- Created all of the relevant migrations for the date_posted removal
- Updated all relevant tests

Fixes #CDD-<ISSUE_ID>

---

## Type of change

Please select the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Tech debt item (this is focused solely on addressing any relevant technical debt)

---

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests at the right levels to prove my change is effective
- [ ] I have added screenshots or screen grabs where appropriate
- [ ] I have added docstrings in the correct style [(google)](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings)
- [x] I have deployed to the AWS environment and tested on machine

### CODE REVIEW

File: cms/common/migrations/0005_remove_commonpage_date_posted.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Generated by Django 5.1.2 on 2024-11-05 13:36
```
Comment (A-Ashiq) [inline cms/common/migrations/0005_remove_commonpage_date_posted.py:1]: So there is another open PR here -> https://github.com/UKHSA-Internal/data-dashboard-api/pull/1913 
which also adds migrations at the same nodes in the migration tree that you have so you're gonna need to co-ordinate between that PR. i.e. one them has to go in first and then the other will need to pull from main and then update the migration refs accordingly

File: tests/unit/caching/test_internal_api_client.py [hunk 2]
```diff
@@ -167,7 +167,7 @@ def test_build_headers(self, force_refresh: bool, cache_check_only: bool):
 
     @pytest.mark.parametrize(
         "additional_query_parameters",
-        ({"fields": "*"}, {"fields": "date_posted"}),
+        ({"fields": "*"}, {}),
```
Comment (A-Ashiq) [inline tests/unit/caching/test_internal_api_client.py:170]: Do we need to test the case of an empty dict i.e. no additional query params here?

File: cms/common/migrations/0005_remove_commonpage_date_posted.py (hunk 1 above)
Comment (luketowell) [reply cms/common/migrations/0005_remove_commonpage_date_posted.py:1]: I will reach out to Phill and see when he will be ready to merge. I also need to coordinate this PR with the FE PR https://github.com/UKHSA-Internal/data-dashboard-frontend/pull/539

File: tests/unit/caching/test_internal_api_client.py (hunk 2 above)
Comment (luketowell) [reply tests/unit/caching/test_internal_api_client.py:170]: I am reading through the test cast and I think it might be worth keeping in the fields and the test for additional query params. I think the test is more for passing additional query params, and "date_posted" is an example of an additional param rather than related to date_posted. 

TLDR: Is it worth reverting this change as I think the test is valid and I got a little trigger happy with removing the contents of the dictionary?

Comment (A-Ashiq) [reply tests/unit/caching/test_internal_api_client.py:170]: Yeah I actually think it's better to keep it as it was, the date_posted is still valid for the whats new child page type

Comment (luketowell) [reply tests/unit/caching/test_internal_api_client.py]: I agree, I have reverted the change back to what it was previously 👍

**Review by A-Ashiq — APPROVED** [review]
